pack/NEONprocIS.base/tests/testthat/test-def.wrte.avro.deve.R

```diff
@@ -1,26 +1,143 @@
+
+#' changelog and author contributions / copyrights
+#'     revised the test
+#'   Mija Choi (2020-04-08)
+#'     add the OS info due to ravro filetype constraint and execute this test only if the OS is Linux based
+#'     Execute the test when the OS is Linux-based, skip otherwise
+##############################################################################################
+
+test_that("   Testing def.wrte.avro.deve.R, definition function. Write AVRO file",
+          {
+            os_type = Sys.info()["sysname"]
+            
+            if (os_type == "Linux")
+            {
+              
+              workingDirPath <- getwd()
+              
+              nameFile <- file.path(workingDirPath, "testdata/HART_data.avro")
+              outFile <- file.path(workingDirPath, "testdata/HART_out.avro")
+              schm <- file.path(workingDirPath, "testdata/HART_data.avsc")
+              nameLib <- file.path(workingDirPath, "ravro.so")
+              # valid data, data_df
+              data_df <- NEONprocIS.base::def.read.avro.deve(NameFile = nameFile, NameLib = nameLib)
+              # invalid data, data
+              data <- data.frame(
+                "source_id" = (c("19963", "19963", "19963", "19963", "19963")),
+                "site_id" = c('HARV', 'HARV', 'HARV', 'HARV', 'HARV'),
+                "readout_time" = as.POSIXct(c(('2019-01-01 00:00:00'),
+                                              ('2019-01-01 00:00:10'),
+                                              ('2019-01-01 00:00:20'),
+                                              ('2019-01-01 00:00:30'),
+                                              ('2019-01-01 00:00:40'))),
+                "resistance" = c(100.0767, 100.0769, 100.0771, 100.0771, 100.0763)
+              )
+ 
+            # Happy path 1: When the minimal parameters,input data, output file and the avro ibrary, are passed in
+            # returns success, 0  
+              cat("\n |=================================================================================|")
+              cat("\n\n   Test 1: When only the input data, output file and the avro ibrary are passed in.")
+              cat("\n   Write successful.")
+              cat("\n |=================================================================================|\n")
+              # reset avro_write_success
+              avro_write_success = -1
+              avro_write_success <- NEONprocIS.base::def.wrte.avro.deve (
+                    data = data_df, 
+                    NameFile = outFile,
+                    NameSchm = NULL, 
+                    NameSpceSchm =  NULL, 
+                    Schm = NULL, 
+                    NameFileSchm = NULL, 
+                    NameLib = 'ravro.so')
+              expect_true (file.exists(outFile) &&avro_write_success == 0 ) 
+              # remove the test ouput file
+              if (file.exists(outFile)) {file.remove(outFile)}
+              # reset
+              writeSuccess = -1
```

Comment:
I think these writeSuccess instances should be avro_write_success. A few more below.

Reply:
Oh, a good catch.  
It was writeSuccess and somehow I introduced avro_write_success from my scratch script.  
Pushed up. Thanks!